### PR TITLE
Full stack/preparation for refactoring data persistence

### PR DESCRIPTION
**Pull request by vanbasten17**
## Description
* Workaround to wait until websocket server and rest server are up and running in local. https://github.com/hubtype/botonic/pull/1866/commits/653cda3acbbcafe919e7ee34e6e919fdf08f33cc
* Fixed annoying warning due to undeprecated extendsDefaultPlugins in webpack's svgo. https://github.com/hubtype/botonic/pull/1866/commits/48448b4241fbdd23344e83afb401d6b06a3e026e
* Moved data-provider from @botonic/api to @botonic/core and update imports. https://github.com/hubtype/botonic/pull/1866/commits/bfe5a3cb6296119543e6e45d5969499774c486b4 and https://github.com/hubtype/botonic/pull/1866/commits/86a4fca8fbc33357e183ff2583529c8cefe33cbd
* Fixed circular dependencies in @botonic/core causing the project to not compile and move legacy types within models folder. https://github.com/hubtype/botonic/pull/1866/commits/7027c634674157973c6d715d57a131afdad239b0
* Update imports from '@botonic/core' in @botonic/api and dev-template (chore(api/botonic-app): rewrite imports from @botonic/core) https://github.com/hubtype/botonic/pull/1866/commits/de1f7795f3ab75e2b846b54b934c67888b6d0a5a

## Context
Preparing core for doing the refactor of data persistence described here: https://linear.app/hubtype/issue/BOT-549/refactor-data-persistance

## Testing

The pull request...

- has unit tests
- has integration tests
- doesn't need tests because... **[provide a description]**

### CODE REVIEW

File: packages/botonic-api/src/rest/routes/events.ts [hunk 1]
```diff
@@ -90,14 +90,14 @@ export default function eventsRouter(args: any): Router {
...
-            ack: MessageEventAck.SENT,
-          })
+          // const newUserEvent = await dp.saveEvent({
+          //   ...parsedUserEvent,
+          //   userId,
+          //   eventId: ulid(),
+          //   createdAt: new Date().toISOString(),
+          //   from: MessageEventFrom.USER,
+          //   ack: MessageEventAck.SENT,
+          // })
```
Comment (asastre) [inline packages/botonic-api/src/rest/routes/events.ts:100]: Remove commented lines?

File: packages/botonic-core/src/data-provider/data-provider.ts [hunk 2]
```diff
@@ -1,13 +1,4 @@
...
-import { LocalDevDataProvider } from './local-data-provider'
-
-export const dataProviderProtocols = {
-  DYNAMO_DB: 'dynamodb',
-  FILE: 'file',
-}
+import { BotonicEvent, User } from '../models'
 
 export interface DataProvider {
   getUsers(limit?: number, offset?: number): User[] | Promise<User[]>
```
Comment (asastre) [inline packages/botonic-core/src/data-provider/data-provider.ts:4]: Does it make sense to have this file with only the interface defined? Would it be better to define this interface in `data-provider/index.t`s file or `data-provider/factory.ts` file?

**Review by asastre — APPROVED** [review]
LGTM!

File: packages/botonic-api/src/rest/routes/events.ts (hunk 1 above)
Comment (vanbasten17) [reply packages/botonic-api/src/rest/routes/events.ts:100]: yes, they are removed in pt.2!

File: packages/botonic-core/src/data-provider/data-provider.ts (hunk 2 above)
Comment (vanbasten17) [reply packages/botonic-core/src/data-provider/data-provider.ts]: yes, maybe it's not that necessary. moved into factory.ts! (I preferred not to move it into index to preserve the same project structure -exporting all members in index- and avoid future problems with circulars deps)

Comment (vanbasten17) [reply packages/botonic-core/src/data-provider/data-provider.ts]: https://github.com/hubtype/botonic/pull/1866/commits/e6729b3139036a4ab1d66c8c7e8c103868883be3

**Review by ericmarcos — APPROVED** [review]

**Pull request merged**